hivemq-edge/src/frontend/src/components/rjsf/Form/error-focus.utils.ts

```diff
@@ -0,0 +1,107 @@
+import { MutableRefObject } from 'react'
+import Form from '@rjsf/core'
+import { RJSFSchema, RJSFValidationError, UiSchema } from '@rjsf/utils'
+import { ChakraRJSFormContext, UITab, UITabIndexed } from '@/components/rjsf/Form/types.ts'
+
+export const _toPath = (path: string) => path?.match(/([^[.\]])+/g)
+
+// From RJSF
+/* istanbul ignore next -- @preserve */
+const focusOnError = (formElement: HTMLFormElement, error: RJSFValidationError) => {
+  // const { idPrefix = 'root', idSeparator = '_' } = props;
+  const idPrefix = 'root',
+    idSeparator = '_'
+
+  const { property } = error
+  if (!property) return
+
+  const path = _toPath(property)
+  if (!path) return
+  if (path[0] === '') {
+    // Most of the time the `.foo` property results in the first element being empty, so replace it with the idPrefix
+    path[0] = 'root'
+  } else {
+    // Otherwise insert the idPrefix into the first location using unshift
+    path.unshift(idPrefix)
+  }
+
+  const elementId = path.join(idSeparator)
+  // let field = formElement.elements[elementId]
+  let field = formElement.querySelector(`#${elementId}`) as HTMLElement
+
+  if (!field) {
+    // if not an exact match, try finding an input starting with the element id (like radio buttons or checkboxes)
+    // field = this.formElement.current.querySelector(`input[id^="${elementId}"`)
+    field = formElement.querySelector(`input[id^="${elementId}"`) as HTMLElement
+  }
+  if (field instanceof HTMLDivElement) {
+    // if the field is a select or other wrapper, find the nested input
+    field = formElement.querySelector<HTMLDivElement>(`#${elementId} input`) as HTMLElement
+  }
+
+  if (field) {
+    field.focus()
+  }
+}
+
+/* istanbul ignore next -- @preserve */
+export const customFocusError = (wrapperRef: MutableRefObject<null>) => (error: RJSFValidationError) => {
+  if (!wrapperRef.current) return
+
+  const rjsForm = wrapperRef.current as Form
+
+  focusOnError(rjsForm.formElement.current, error)
+}
+
+export const isNumeric = (element: string) => /^\d+$/.test(element)
+export const deepGet = (obj: object, keys: (string | number)[]) => {
```

Comment:
genuine question: why not using [lodash's get()](https://lodash.com/docs/4.17.15#get)?

Reply:
good question. Force of habits when lodash was a bulky lib, preferring plain alternatives especially if only one function needed